server/utils/bot/flower.go

```diff
@@ -0,0 +1,105 @@
+package bot
+
+import "github.com/mattermost/mattermost-plugin-mscalendar/server/utils/flow"
+
+type Flower interface {
+	Start(userID string) error
+	NextStep(userID string, value bool) error
+	Cancel(userID string) error
+}
+
+func (bot *bot) Start(userID string) error {
+	err := bot.setFlowStep(userID, 0)
+	if err != nil {
+		return err
+	}
+	return bot.processStep(userID, bot.flow.Step(0), 0)
+}
+
+func (bot *bot) NextStep(userID string, value bool) error {
+	step, err := bot.getFlowStep(userID)
+	if err != nil {
+		return err
+	}
+
+	skip := bot.flow.Step(step).ShouldSkip(value)
+	step += 1 + skip
+	if step >= bot.flow.Length() {
+		bot.removeFlowStep(userID)
+		bot.flow.FlowDone(userID)
+		return nil
+	}
+
+	err = bot.setFlowStep(userID, step)
+	if err != nil {
+		return err
+	}
+
+	return bot.processStep(userID, bot.flow.Step(step), step)
+}
+
+func (bot *bot) Cancel(userID string) error {
+	stepIndex, err := bot.getFlowStep(userID)
+	if err != nil {
+		return err
+	}
+
+	step := bot.flow.Step(stepIndex)
+	if step == nil {
+		return nil
+	}
+
+	postID, err := bot.flowStore.GetPostID(userID, step.PropertyName())
+	if err != nil {
+		return err
+	}
+
+	err = bot.DeletePost(postID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (bot *bot) setFlowStep(userID string, step int) error {
+	// TODO: Use KVStore for HA
```

Comment:
I assume this will be covered in this PR? I'd be fine with deferring to a ticket if you want to get this merged sooner.

Reply:
My plan is to have this covered in this PR, but I wanted to confirm first that the new shape of the flow was interesting. I will take a look to have this done tomorrow.